### PR TITLE
Include error indicator field in HWP ManifestDb

### PR DESCRIPTION
**Pull request by yukisakurai**
Added a field (`hwp_err`) in HWP ManifestDb to indicate when the HWP angle is all zeros, whether it is a calculation failure or just a lack of L2 HWP data.
Motivated by this issue: #669

### CODE REVIEW

File: sotodlib/hwp/g3thwp.py [hunk 1]
```diff
@@ -643,18 +643,20 @@ def _set_empty_axes(self, aman):
...
-    def _write_empty_solution_h5(self, tod, output=None, h5_address=None):
-
+    def _write_empty_solution_h5(self, tod, output=None, h5_address=None, err=False):
+        
         logger.info('Writing empty solutions')
         # metadata loader requires a dets axis
         aman = sotodlib.core.AxisManager(tod.dets, tod.samps)
         self._set_empty_axes(aman)
         aman.timestamps[:] = tod.timestamps
+        if err: aman.hwp_err[:] = np.ones(len(tod.timestamps))
```
Comment (ykyohei) [inline sotodlib/hwp/g3thwp.py:657]: I prefer something like
`aman.hwp_err[:] = np.full_like(tod.timestamps, err, dtype=bool)`

hwp_err is misleading as hwp angle uncertainty, so should we change to err_flag or err_msg?

But I think the error flag can be just a single bool value not an array.
Can we write a single bool value error flag to the h5 metadata file instead?
Or saving simple message like "No data", "data is too short", "Error in hwp angle calculation" would be more helpful?

File: sotodlib/hwp/g3thwp.py [hunk 2]
```diff
@@ -643,18 +643,20 @@ def _set_empty_axes(self, aman):
...
         aman.wrap_new('hwp_rate', shape=('samps', ), dtype=np.float16)
         aman.wrap_new('eval', shape=('samps', ), dtype=bool)
+        aman.wrap_new('hwp_err', shape=('samps', ), dtype=bool)
 
         return
 
-    def _write_empty_solution_h5(self, tod, output=None, h5_address=None):
-
+    def _write_empty_solution_h5(self, tod, output=None, h5_address=None, err=False):
+
```
Comment (ykyohei) [inline sotodlib/hwp/g3thwp.py:651]: Please delete unnecessary spaces. There is another one.

**Review by ykyohei — CHANGES_REQUESTED** [review]

**Comment by yukisakurai** [comment]
Changed from boolean array to single logger message, `hwp_logger`.
The same stuff is happening with the `eval` flag, so I renamed it to `hwp_angle_ver2_flag` and changed to a single boolean.

File: sotodlib/site_pipeline/make_hwp_solutions.py [hunk 3]
```diff
@@ -18,6 +18,7 @@
 from sotodlib.hwp.g3thwp import G3tHWP
 from sotodlib.site_pipeline import util
 default_logger = util.init_logger(__name__, 'make-hwp-solutions: ')
+print(sotodlib.hwp.g3thwp.__file__)
```
Comment (ykyohei) [inline sotodlib/site_pipeline/make_hwp_solutions.py:21]: This line looks unnecessary.

File: sotodlib/hwp/g3thwp.py [hunk 4]
```diff
@@ -737,15 +736,19 @@ def write_solution_h5(self, tod, output=None, h5_address=None):
         try:
             solved = self.analyze(data, mod2pi=False)
         except Exception as e:
-            logger.error(f"Exception '{e}' thrown while calculating HWP angle. Encoder signal might have too much noise.")
+            logger.error(f"Exception '{e}' thrown while calculating HWP angle. Angle calculation failed.")
+            self._write_solution_h5_logger = 'Angle calculation failed'
             self._write_empty_solution_h5(tod, output, h5_address)
             return
-
+
```
Comment (ykyohei) [inline sotodlib/hwp/g3thwp.py:743]: unnecessary space

File: sotodlib/hwp/g3thwp.py [hunk 5]
```diff
@@ -723,11 +720,13 @@ def write_solution_h5(self, tod, output=None, h5_address=None):
             data = self.load_data(start, end)
         except Exception as e:
             logger.error(f"Exception '{e}' thrown while loading HWP data. The specified encoder field is missing.")
+            self._write_solution_h5_logger = 'HWP data too short'
```
Comment (ykyohei) [inline sotodlib/hwp/g3thwp.py:723]: This logger should be "Error in loading HWPdata"?

**Review by ykyohei — COMMENTED** [review]
Thank you. Overall looks good to me. Made some minor comments.

File: sotodlib/hwp/g3thwp.py (hunk 5 above)
Comment (yukisakurai) [reply sotodlib/hwp/g3thwp.py:723]: This is not an error, just a short data length and no need to debug, so I prefer to leave it.

**Review by ykyohei — APPROVED** [review]

**Pull request merged**